pkg/reconciler/taskrun/taskrun.go

```diff
@@ -629,8 +629,13 @@ func (c *Reconciler) createPod(ctx context.Context, tr *v1beta1.TaskRun, rtr *re
 		return nil, fmt.Errorf("translating TaskSpec to Pod: %w", err)
 	}
 
-	return c.KubeClientSet.CoreV1().Pods(tr.Namespace).Create(pod)
-
+	pod, err = c.KubeClientSet.CoreV1().Pods(tr.Namespace).Create(pod)
+	if err == nil && willOverwritePodSetAffinity(tr) {
+		if recorder := controller.GetEventRecorder(ctx); recorder != nil {
+			recorder.Eventf(tr, corev1.EventTypeWarning, "PodAffinityOverwrite", "Pod template affinity is overwritten by affinity assistant for pod %q", pod.Name)
```

Comment:
I would introduced `const` here for `PodAffinityOverwrite` but going through the reviews from @jlpettersson on the same 🙃 , looks good for now, can be converted to `const` if needed in future